acp/identity.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	cosmosSecp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/sourcenetwork/immutable"
+)
+
+// NoIdentity specifies an anonymous actor.
+var NoIdentity = immutable.None[Identity]()
+
+// Identity uniquely identifies an acp actor.
+type Identity interface {
+	// PublicKey returns the public key of the identity.
+	PublicKey() *secp256k1.PublicKey
+	// Address returns the bech32 address of the identity.
```

Comment:
suggestion: The callers of this function (with a lazy ctrl+f in the browser) are in relatively generalized parts of the codebase where maintainers do not need to understand or know what a `bech32 address` and why that is significant.

At the moment this comment/documentation does not describe if or why it being `bech32` is significant, and the calling sites appear to be ignorant of that too.

If its `bech32`ness is important, please describe why.  If it is not I suggest either removing it from the comment, or reducing its importance by moving it onto a secondary line (e.g. below).
```
// Address returns the address of the identity.  // the address where (maybe also note if this is in sourcehub/keyring/whereever)
//
// note on it being in bech32 format
```